python/hard_clone_workspace.py

```diff
@@ -0,0 +1,208 @@
+        self.dest_workspace = dest_workspace
+
+    def run(self) -> None:
+        # Get the source workspace ACLs and entities
+        src_workspace_acls = self.src_workspace.get_workspace_acl()
+        # Convert the source workspace ACLs to a list of dictionaries
+        src_workspace_acls_list = [
+            {
+                "email": user,
+                "accessLevel": user_acl["accessLevel"],
+                "canShare": user_acl["canShare"],
+                "canCompute": user_acl["canCompute"]
+            }
+            for user, user_acl in src_workspace_acls["acl"].items()
+        ]
+        self.dest_workspace.update_multiple_users_acl(acl_list=src_workspace_acls_list)
+
+
+if __name__ == '__main__':
+    args = get_args()
+    source_billing_project = args.source_billing_project
+    source_workspace_name = args.source_workspace_name
+    dest_billing_project = args.dest_billing_project
+    dest_workspace_name = args.dest_workspace_name
+    allow_already_created = args.allow_already_created
+    workers = args.workers
+    extensions_to_ignore = args.extensions_to_ignore.split(",") if args.extensions_to_ignore else None
+
+    token = Token(cloud=GCP)
+    request_util = RunRequest(token=token)
+    # Initialize the source Terra workspace classes
+    src_workspace = TerraWorkspace(
+        billing_project=source_billing_project,
+        workspace_name=source_workspace_name,
+        request_util=request_util
+    )
+    # Initialize the destination Terra workspace classes
+    dest_workspace = TerraWorkspace(
+        billing_project=dest_billing_project,
+        workspace_name=dest_workspace_name,
+        request_util=request_util
+    )
+
+    # Get the source workspace info
+    src_workspace_info = src_workspace.get_workspace_info()
+    src_auth_domain = src_workspace_info["workspace"]["authorizationDomain"]
+    src_bucket = src_workspace_info["workspace"]["bucketName"]
+
+    # Removing library attributes from the source workspace attributes
+    # This is specific to published firecloud info which this does not support
+    src_attributes = {
+        k: v
+        for k, v in src_workspace_info["workspace"]["attributes"].items()
+        if not k.startswith('library:')
+    }
+
+    # Create the destination workspace
+    dest_workspace.create_workspace(
+        attributes=src_attributes,
+        auth_domain=src_auth_domain,
+        continue_if_exists=allow_already_created
+    )
+    dest_workspace_info = dest_workspace.get_workspace_info()
+    dest_bucket = dest_workspace_info["workspace"]["bucketName"]
+
+    # Get source workspace metadata
+    tsvs_to_upload = CreateEntityTsv(
+        src_bucket=src_bucket,
+        dest_bucket=dest_bucket,
+        source_workspace=src_workspace
+    ).run()
+
+    for tsv in tsvs_to_upload:
+        logging.info(f"Uploading {tsv} to destination workspace")
+        dest_workspace.upload_data_to_workspace_table(entities_tsv=tsv)
```

Comment:
Can we rename `upload_data_to_workspace_table` to `upload_metadata_to_workspace_table`? I think it makes it just a little more clear what it's doing